v2/app/build.gradle.kts

```diff
@@ -0,0 +1,77 @@
+plugins {
+    alias(libs.plugins.android.application)
+    alias(libs.plugins.kotlin.kapt)
+    alias(libs.plugins.kotlin.android)
+    alias(libs.plugins.kotlin.compose)
+    alias(libs.plugins.kotlin.serialization)
+    alias(libs.plugins.google.services)
+    alias(libs.plugins.google.hilt)
+}
+
+android {
+    namespace = "com.google.firebase.example.makeitso"
+    compileSdk = 34
+
+    defaultConfig {
+        applicationId = "com.google.firebase.example.makeitso"
+        minSdk = 26
+        targetSdk = 34
+        versionCode = 1
+        versionName = "1.0"
+
+        testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
+    }
+
+    buildTypes {
+        release {
+            isMinifyEnabled = false
+            proguardFiles(
+                getDefaultProguardFile("proguard-android-optimize.txt"),
+                "proguard-rules.pro"
+            )
+        }
+    }
+    compileOptions {
+        sourceCompatibility = JavaVersion.VERSION_1_8
+        targetCompatibility = JavaVersion.VERSION_1_8
+    }
+    kotlinOptions {
+        jvmTarget = "1.8"
+    }
+    buildFeatures {
+        compose = true
+    }
+    kapt {
+        correctErrorTypes = true
+    }
+}
+
+dependencies {
+    implementation(libs.androidx.core.ktx)
+    implementation(libs.androidx.lifecycle.runtime.ktx)
+    implementation(libs.androidx.activity.compose)
+    implementation(platform(libs.androidx.compose.bom))
+    implementation(libs.androidx.ui)
+    implementation(libs.androidx.ui.graphics)
+    implementation(libs.androidx.ui.tooling.preview)
+    implementation(libs.androidx.material3)
+    implementation(libs.androidx.navigation.compose)
+    implementation(libs.androidx.hilt.navigation.compose)
+    implementation(libs.kotlinx.serialization.json)
+    implementation(libs.hilt.android)
+    kapt(libs.hilt.android.compiler)
```

Comment:
Hilt added support for [KSP](https://github.com/google/ksp) in [version 1.1.0](https://developer.android.com/jetpack/androidx/releases/hilt#hilt_version_110_2) - I'm surprised that the [docs](https://developer.android.com/training/dependency-injection/hilt-android) haven't been updated yet.

I would suggest we use KSP instead of KAPT for Make it so V2 - KSP can provide up to 2x faster build speed, and the migration effort should be minimal (see the [migration guide](https://developer.android.com/build/migrate-to-ksp)).

Reply:
Oh this is great! Good catch, migrated to KSP.